### DISCUSSION TITLE
X-API-KEY via ENV

### DISCUSSION THREAD

**Discussion by Kulikovpavel** [Stirling-Tools/Stirling-PDF#679] [Ideas]
It would be awesome to have the option to set up a single API key for all endpoints without relying on a user's system, just with ENV variable
What do you think?

**Comment by Frooodle** [comment]
Can you explain the benefit of this?
Also to be honest i have never liked any passwords in env either as its less secure

**Reply by Kulikovpavel** [reply]
To use image in public google cloud run / K8 under x-api-key protection without working with manual user setup.

Cloudrun is stateless, so it is impossible to make it work with current flow when you need to open user page/enter passwords/change api-key and safe this data in the local file, which will be deleted on restart.

Also, maybe you can run it in DOCKER_ENABLE_SECURITY=false mode, without the need to download the secure JAR (startup time/external point of failure)
